server/events.go

```diff
@@ -222,6 +216,59 @@ type serverUpdate struct {
 	ltime time.Time
 }
 
+const unknownTypedMessageType = "io.nats.unknown_message"
+
+func setTypedEvent(e TypedEvent, idgen ...func() string) TypedEvent {
```

Comment:
while this is a pain to maintain, its a lot less painful than having to remember which constant to use and fill in the various TypedEvent fields every time one wants to send an advisory.

If you like this pattern I can do the same with the API stuff where we have all these

```go
var resp = JSApiAccountInfoResponse{ApiResponse: ApiResponse{Type: JSApiAccountInfoResponseType}}
```

that you did not like.

Reply:
Might be a better approach.

Reply:
OK, will do a follow up PR for those.

Reply:
I seem to prefer after looking at actual code the old version.

Reply:
OK :) will remove